.github/workflows/deploy-to-aks.yaml

```
@@ -81,7 +96,11 @@ jobs:
 
       - name: Deploy and assert kubeflow-mlflow solution
         run: |
-          tox -c ./modules/kubeflow-mlflow -vve test_deployment -- -vv -s
+          if [[ '${{ github.event_name }}' != 'pull_request' ]]; then
+            git checkout 'track/${{ env.CKF_VERSION }}'
+          fi
```

question: I'm not sure why we only `checkout` on pull requests. If I understand correctly, this workflow only runs on workflow dispatch, so this never occurs.
```suggestion
          if [[ '${{ github.event_name }}' != 'pull_request' ]]; then
            git checkout 'track/${{ env.CKF_VERSION }}'
          fi
```

---

> I'm not sure why we only checkout on pull requests.

On pull request we don't checkout, since we want to use the tip of the the PR to indeed tests the current changes. On the other hand, when we dispatch and run the action manually, on the other hand, we want to switch to the track provided as input in the dispatch. 

is this clearer?